From a133d644138b6018b5df2bcb4bba07211a9ac5b7 Mon Sep 17 00:00:00 2001
From: "Benjamin A. Beasley" <code@musicinmybrain.net>
Date: Wed, 17 Jan 2024 21:43:39 -0500
Subject: [PATCH 1/3] Add some missing #includes for absl/strings/str_cat.h

Partial backport of a giant IWYU commit:

https://github.com/grpc/grpc/commit/8174a75079bc00aec4f80e3cd9f91a7042073ca8

Needed for abseil-cpp-20240116.rc1 compatibility.
---
 test/core/address_utils/sockaddr_utils_test.cc | 1 +
 test/core/channel/call_finalization_test.cc    | 1 +
 test/core/json/json_test.cc                    | 1 +
 3 files changed, 3 insertions(+)

diff --git a/test/core/address_utils/sockaddr_utils_test.cc b/test/core/address_utils/sockaddr_utils_test.cc
index 3ecb5a7f82..e071d94e41 100644
--- a/test/core/address_utils/sockaddr_utils_test.cc
+++ b/test/core/address_utils/sockaddr_utils_test.cc
@@ -24,6 +24,7 @@
 #include <sys/un.h>
 #endif
 
+#include "absl/strings/str_cat.h"
 #include <gtest/gtest.h>
 
 #include <grpc/grpc.h>
diff --git a/test/core/channel/call_finalization_test.cc b/test/core/channel/call_finalization_test.cc
index 61d92c5f56..0ff7a00cff 100644
--- a/test/core/channel/call_finalization_test.cc
+++ b/test/core/channel/call_finalization_test.cc
@@ -14,6 +14,7 @@
 
 #include "src/core/lib/channel/call_finalization.h"
 
+#include "absl/strings/str_cat.h"
 #include <gtest/gtest.h>
 
 #include "src/core/lib/resource_quota/resource_quota.h"
diff --git a/test/core/json/json_test.cc b/test/core/json/json_test.cc
index 232caca9ba..ff26b7ee75 100644
--- a/test/core/json/json_test.cc
+++ b/test/core/json/json_test.cc
@@ -20,6 +20,7 @@
 
 #include <string.h>
 
+#include "absl/strings/str_cat.h"
 #include <gmock/gmock.h>
 #include <gtest/gtest.h>
 
-- 
2.43.0


From 9f2762981680094d97a6092e76c8b94309df9512 Mon Sep 17 00:00:00 2001
From: Esun Kim <veblush@google.com>
Date: Thu, 14 Sep 2023 17:25:41 -0700
Subject: [PATCH 2/3] [Fix] Added missing #include (#34359)

One more missing #include
---
 test/core/tsi/crl_ssl_transport_security_test.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/test/core/tsi/crl_ssl_transport_security_test.cc b/test/core/tsi/crl_ssl_transport_security_test.cc
index 5e4dc1b57e..dd33173abf 100644
--- a/test/core/tsi/crl_ssl_transport_security_test.cc
+++ b/test/core/tsi/crl_ssl_transport_security_test.cc
@@ -19,6 +19,8 @@
 #include <gmock/gmock.h>
 #include <gtest/gtest.h>
 
+#include "absl/strings/str_cat.h"
+
 #include <grpc/grpc.h>
 #include <grpc/support/alloc.h>
 #include <grpc/support/log.h>
-- 
2.43.0


From 8a53ca0b02de4074763ec970c85ab9594d99532c Mon Sep 17 00:00:00 2001
From: "Benjamin A. Beasley" <code@musicinmybrain.net>
Date: Thu, 18 Jan 2024 08:07:10 -0500
Subject: [PATCH 3/3] Add abseil includes in test/cpp/end2end/xds/xds_server.h
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This follows the IWYU “philosophy,” but the necessary includes were
found and added manually.

Needed for compatibility with abseil-cpp 20240116.rc1.
---
 test/cpp/end2end/xds/xds_server.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/test/cpp/end2end/xds/xds_server.h b/test/cpp/end2end/xds/xds_server.h
index 066629b101..4499af1a8d 100644
--- a/test/cpp/end2end/xds/xds_server.h
+++ b/test/cpp/end2end/xds/xds_server.h
@@ -26,6 +26,10 @@
 #include <gmock/gmock.h>
 #include <gtest/gtest.h>
 
+#include "absl/status/status.h"
+#include "absl/strings/numbers.h"
+#include "absl/strings/string_view.h"
+#include "absl/time/time.h"
 #include "absl/types/optional.h"
 
 #include <grpc/support/log.h>
-- 
2.43.0

